File: week2/furkanunsalan/155-min-stack.py
class MinStack:

    def __init__(self):
        self.A = []
        self.min = []
    def push(self, val: int) -> None:
        self.A.append(val)
        self.min.append(val if not self.min else min(val, self.min[-1]))
    def pop(self) -> None:
        self.A.pop()
        self.min.pop()
    def top(self) -> int:
        return self.A[-1]
    def getMin(self) -> int:
        return self.min[-1]

File: week2/Eren-Taskiran/155-min-stack.java
class MinStack {
    Node node;
    public MinStack() {
    }
    
    public void push(int val) {
       Node newnode = new Node(val,node);
       node = newnode;
    }
    
    public void pop() {
        node=node.prev;
    }
    
    public int top() {
        return node.value;
    }
    
    public int getMin() {
        return node.min;
    }
    class Node{
        int value;
        Node prev;
        int min;
        Node(int value, Node prev){
            this.value=value;
            this.prev=prev;
            if( prev!=null && value>prev.min ){
                this.min =prev.min;
            }else{
                this.min=value;
            }
        }
    }
}